### DISCUSSION TITLE
Is my IM borked or am I stupid?

### DISCUSSION THREAD

**Discussion by tifrel** [ImageMagick/ImageMagick#1931] [General] [answered]
I don't know if this is a new bug or a widely known issue, but either my `magick`/`convert` behave really badly, or I didn't get how it's supposed to work.

So let's do some examples:
    
    convert in.png -alpha set -region <geom> -alpha transparent out.png

Which should do the trick according to [IMv6 docs](https://www.imagemagick.org/Usage/masking/#regions). I run IMv7, so that might have changed. My image however, has not, at least not visually. It's grown by 120 kB, and `cmp` reports a difference quite early (byte 2843, from 700 kB on the original image). Ok so I thought I might have missed the region, however:

    convert in.png -region <geom> -negate out.png

applies the changes: alpha -> white, black -> alpha. Cannot report on other colors because there are none. Seems legit, using magick for the 3rd time or so in my life, I cannot tell if this is the expected behavior. Now I know I hit the region. So I try to put any color on the region, let's say blue:

    convert in.png -region <geom> -fill blue -colorize '100%' out.png

It is now that the region disappears. I also get blue artifacts at the borders between alpha and non-alpha elsewhere in the picture. Can be solved by using `-colorize '0%'`, but this seriously wtf'ed me.

So is this expected behavior and did I just not understand shit about `magick` or can anyone lift the mystery for me?

IM Version: 7.0.10-7 Q16 x86_64 2020-04-20
OS: 4.19.117-1-MANJARO

**Comment by fmw42** [comment]
-region works differently in Imagemagick 7. I do not recommend using it. I would create a rectangular mask.

**Reply by tifrel** [reply]
As I said, I am rather new to IM, could you elaborate, in the best case in terms of commands needed, what needs to be done to get this to work 'the proper way'

**Reply by fmw42** [reply]
For new users, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli

**Comment by fmw42** [comment, marked as answer]
If you want to make a rectangular region in an image transparent in Imagemagick 7, then here is one way.

- read the input
- clone it and fill it over with white
- draw a black rectangle somewhere in it
- put the black and white image into the alpha channel of the input
- save the output

![lena](https://user-images.githubusercontent.com/13458218/80241476-806a2b00-8618-11ea-8857-3204524052a1.png)

Unix Syntax:

```
magick lena.png \
\( -clone 0 -fill white -colorize 100 \
-fill black -draw "rectangle 50,50 100,100" \) \
-alpha off -compose copy_opacity -composite \
lena_transparent_region.png

```
![lena_transparent_region](https://user-images.githubusercontent.com/13458218/80241523-911aa100-8618-11ea-8409-9dcccc31e339.png)


See
https://imagemagick.org/Usage/draw/#primitives
https://imagemagick.org/Usage/compose/#copyopacity

**Reply by tifrel** [reply]
Thanks a lot!
